### PR TITLE
Update text docs

### PR DESCRIPTION
**Pull request by reaganjlee**
## What does this PR do?

Fixes Text docs as a part of #1365 

## Before submitting

- [x] Was this **discussed/approved** via a Github issue? (no need for typos and docs improvements)
- [x] Did you read the [contributor guideline](https://github.com/Lightning-AI/metrics/blob/master/.github/CONTRIBUTING.md), Pull Request section?
- [ ] Did you make sure to **update the docs**?
- [ ] Did you write any new **necessary tests**?

# Some Notes

- Squad.py, 
    - There was a code example, which I wasn't sure if we should keep or not. I left it in for now. 
    - I left the method-specific raise within the method docstring
- sacre_bleu.py, Wasn't sure where to put an additional reference that didn't have an in-line citation, left it on the bottom under "additional citations" for now
- Ter.py, references a specific file within a github repo that was already cited as the repo already. Wasn't sure if this was important, left them as two citations just in case 

No shape info, so no commit needed for that

### CODE REVIEW

File: src/torchmetrics/text/bert.py [hunk 1]
```diff
@@ -47,6 +47,14 @@ class BERTScore(Metric):
 
     This implemenation follows the original implementation from `BERT_score`_.
 
+    As input to 'update' and 'forward' the metric accepts the following input:
+
+    - ``preds``: An iterable of predicted sentences
+    - ``target``: An iterable of reference sentences
+
+    As output of 'compute' and 'forward' the metric returns a Python dictionary containing the keys `precision`,
+    `recall` and `f1` with corresponding values.
```
Comment (Borda) [inline src/torchmetrics/text/bert.py:56]:
```suggestion
    As output of 'compute' and 'forward' the metric returns a Python dictionary containing the keys ``precision``,
    ``recall`` and ``f1`` with corresponding values.
```

**Review by Borda — APPROVED** [review]

**Review by SkafteNicki — CHANGES_REQUESTED** [review]
Just blocking until output formatting is also updated, discussed on slack

File: src/torchmetrics/text/bleu.py [hunk 2]
```diff
@@ -84,12 +86,7 @@ def __init__(
         self.add_state("denominator", torch.zeros(self.n_gram), dist_reduce_fx="sum")
 
     def update(self, preds: Sequence[str], target: Sequence[Sequence[str]]) -> None:
-        """Compute Precision Scores.
-
-        Args:
-            preds: An iterable of machine translated corpus
-            target: An iterable of iterables of reference corpus
-        """
+        """Compute Precision Scores."""
```
Comment (SkafteNicki) [inline src/torchmetrics/text/bleu.py:89]:
```suggestion
        """Update state with predictions and targets."""
```

File: src/torchmetrics/text/cer.py [hunk 3]
```diff
@@ -70,20 +77,11 @@ def __init__(
         self.add_state("total", tensor(0, dtype=torch.float), dist_reduce_fx="sum")
 
     def update(self, preds: Union[str, List[str]], target: Union[str, List[str]]) -> None:  # type: ignore
-        """Store references/predictions for computing Character Error Rate scores.
-
-        Args:
-            preds: Transcription(s) to score as a string or list of strings
-            target: Reference(s) for each speech input as a string or list of strings
-        """
+        """Store references/predictions for computing Character Error Rate scores."""
```
Comment (SkafteNicki) [inline src/torchmetrics/text/cer.py:80]:
```suggestion
        """Update state with predictions and targets."""
```

File: src/torchmetrics/text/chrf.py [hunk 4]
```diff
@@ -125,12 +130,7 @@ def __init__(
             self.add_state("sentence_chrf_score", [], dist_reduce_fx="cat")
 
     def update(self, preds: Sequence[str], target: Sequence[Sequence[str]]) -> None:
-        """Compute Precision Scores.
-
-        Args:
-            preds: An iterable of hypothesis corpus.
-            target: An iterable of iterables of reference corpus.
-        """
+        """Compute Precision Scores."""
```
Comment (SkafteNicki) [inline src/torchmetrics/text/chrf.py:133]:
```suggestion
        """Update state with predictions and targets."""
```

File: src/torchmetrics/text/eed.py [hunk 5]
```diff
@@ -91,12 +93,7 @@ def update(
         preds: Union[str, Sequence[str]],
         target: Sequence[Union[str, Sequence[str]]],
     ) -> None:
-        """Update ExtendedEditDistance statistics.
-
-        Args:
-            preds: An iterable of hypothesis corpus
-            target: An iterable of iterables of reference corpus
-        """
+        """Update ExtendedEditDistance statistics."""
```
Comment (SkafteNicki) [inline src/torchmetrics/text/eed.py:96]:
```suggestion
        """Update state with predictions and targets."""
```

File: src/torchmetrics/text/infolm.py [hunk 6]
```diff
@@ -144,14 +151,7 @@ def __init__(
...
     def update(self, preds: Union[str, Sequence[str]], target: Union[str, Sequence[str]]) -> None:
-        """Update the metric state by a tokenization of ``preds`` and ``target`` sentencens.
-
-        Args:
-            preds:
-            An iterable of hypothesis corpus.
-        target:
-            An iterable of reference corpus.
-        """
+        """Update the metric state by a tokenization of ``preds`` and ``target`` sentencens."""
```
Comment (SkafteNicki) [inline src/torchmetrics/text/infolm.py:154]:
```suggestion
        """Update state with predictions and targets."""
```

File: src/torchmetrics/text/mer.py [hunk 7]
```diff
@@ -72,12 +77,7 @@ def update(  # type: ignore
         preds: Union[str, List[str]],
         target: Union[str, List[str]],
     ) -> None:
-        """Store references/predictions for computing Match Error Rate scores.
-
-        Args:
-            preds: Transcription(s) to score as a string or list of strings
-            target: Reference(s) for each speech input as a string or list of strings
-        """
+        """Store references/predictions for computing Match Error Rate scores."""
```
Comment (SkafteNicki) [inline src/torchmetrics/text/mer.py:80]:
```suggestion
        """Update state with predictions and targets."""
```

File: src/torchmetrics/text/perplexity.py [hunk 8]
```diff
@@ -59,22 +68,11 @@ def __init__(
...
     def update(self, preds: Tensor, target: Tensor) -> None:
-        """Compute and store intermediate statistics for Perplexity.
-
-        Args:
-            preds:
-                Probabilities assigned to each token in a sequence with shape [batch_size, seq_len, vocab_size].
-            target:
-                Ground truth values with a shape [batch_size, seq_len].
-        """
+        """Compute and store intermediate statistics for Perplexity."""
```
Comment (SkafteNicki) [inline src/torchmetrics/text/perplexity.py:71]:
```suggestion
        """Update state with predictions and targets."""
```

File: src/torchmetrics/text/rouge.py [hunk 9]
```diff
@@ -129,12 +137,7 @@ def __init__(
     def update(
         self, preds: Union[str, Sequence[str]], target: Union[str, Sequence[str], Sequence[Sequence[str]]]
     ) -> None:
-        """Compute rouge scores.
-
-        Args:
-            preds: An iterable of predicted sentences or a single predicted sentence.
-            target: An iterable of target sentences or an iterable of target sentences or a single target sentence.
-        """
+        """Compute rouge scores."""
```
Comment (SkafteNicki) [inline src/torchmetrics/text/rouge.py:140]:
```suggestion
        """Update state with predictions and targets."""
```

File: src/torchmetrics/text/sacre_bleu.py [hunk 10]
```diff
@@ -98,12 +103,7 @@ def __init__(
         self.tokenizer = _SacreBLEUTokenizer(tokenize, lowercase)
 
     def update(self, preds: Sequence[str], target: Sequence[Sequence[str]]) -> None:
-        """Compute Precision Scores.
-
-        Args:
-            preds: An iterable of machine translated corpus
-            target: An iterable of iterables of reference corpus
-        """
+        """Compute Precision Scores."""
```
Comment (SkafteNicki) [inline src/torchmetrics/text/sacre_bleu.py:106]:
```suggestion
        """Update state with predictions and targets."""
```

File: src/torchmetrics/text/squad.py [hunk 11]
```diff
@@ -67,38 +104,6 @@ def __init__(
     def update(self, preds: PREDS_TYPE, target: TARGETS_TYPE) -> None:
         """Compute F1 Score and Exact Match for a collection of predictions and references.
```
Comment (SkafteNicki) [inline src/torchmetrics/text/squad.py:105]:
```suggestion
        """Update state with predictions and targets."""
```

File: src/torchmetrics/text/ter.py [hunk 12]
```diff
@@ -84,12 +89,7 @@ def __init__(
             self.add_state("sentence_ter", [], dist_reduce_fx="cat")
 
     def update(self, preds: Union[str, Sequence[str]], target: Sequence[Union[str, Sequence[str]]]) -> None:
-        """Update TER statistics.
-
-        Args:
-            preds: An iterable of hypothesis corpus.
-            target: An iterable of iterables of reference corpus.
-        """
+        """Update TER statistics."""
```
Comment (SkafteNicki) [inline src/torchmetrics/text/ter.py:92]:
```suggestion
        """Update state with predictions and targets."""
```

File: src/torchmetrics/text/wer.py [hunk 13]
```diff
@@ -67,20 +73,11 @@ def __init__(
         self.add_state("total", tensor(0, dtype=torch.float), dist_reduce_fx="sum")
 
     def update(self, preds: Union[str, List[str]], target: Union[str, List[str]]) -> None:  # type: ignore
-        """Store references/predictions for computing Word Error Rate scores.
-
-        Args:
-            preds: Transcription(s) to score as a string or list of strings
-            target: Reference(s) for each speech input as a string or list of strings
-        """
+        """Store references/predictions for computing Word Error Rate scores."""
```
Comment (SkafteNicki) [inline src/torchmetrics/text/wer.py:76]:
```suggestion
        """Update state with predictions and targets."""
```

File: src/torchmetrics/text/wil.py [hunk 14]
```diff
@@ -66,21 +73,12 @@ def __init__(
         self.add_state("preds_total", tensor(0.0), dist_reduce_fx="sum")
 
     def update(self, preds: Union[str, List[str]], target: Union[str, List[str]]) -> None:
-        """Store predictions/references for computing Word Information Lost scores.
-
-        Args:
-            preds: Transcription(s) to score as a string or list of strings
-            target: Reference(s) for each speech input as a string or list of strings
-        """
+        """Store predictions/references for computing Word Information Lost scores."""
```
Comment (SkafteNicki) [inline src/torchmetrics/text/wil.py:76]:
```suggestion
        """Update state with predictions and targets."""
```

File: src/torchmetrics/text/wip.py [hunk 15]
```diff
@@ -65,21 +74,12 @@ def __init__(
         self.add_state("preds_total", tensor(0.0), dist_reduce_fx="sum")
 
     def update(self, preds: Union[str, List[str]], target: Union[str, List[str]]) -> None:
-        """Store predictions/references for computing word Information Preserved scores.
-
-        Args:
-            preds: Transcription(s) to score as a string or list of strings
-            target: Reference(s) for each speech input as a string or list of strings
-        """
+        """Store predictions/references for computing word Information Preserved scores."""
```
Comment (SkafteNicki) [inline src/torchmetrics/text/wip.py:77]:
```suggestion
        """Update state with predictions and targets."""
```

**Review by SkafteNicki — APPROVED** [review]

**Pull request merged**